contracts/gov/user-keeper/GovUserKeeper.sol

```diff
@@ -261,26 +261,14 @@ contract GovUserKeeper is IGovUserKeeper, OwnableUpgradeable, ERC721HolderUpgrad
         uint256 currentPowerSnapshotId = ++_latestPowerSnapshotId;
         NFTSnapshot storage snapshot = nftSnapshot[currentPowerSnapshotId];
 
-        uint256 supply = nftInfo.totalSupply == 0
-            ? nftContract.totalSupply()
-            : nftInfo.totalSupply;
-        uint256 totalNftsPower;
-
-        if (!nftInfo.isSupportPower) {
-            totalNftsPower = supply;
+        if (nftInfo.isSupportPower) {
+            snapshot.totalNftsPower = nftContract.totalPower();
         } else {
-            for (uint256 i; i < supply; i++) {
-                uint256 tokenId = nftContract.tokenByIndex(i);
-                (uint256 solePower, uint256 collateral) = nftContract.recalculateNftPower(tokenId);
-                uint256 power = solePower + collateral;
-
-                snapshot.nftPower[tokenId] = power;
-                totalNftsPower += power;
-            }
+            snapshot.totalNftsPower = nftInfo.totalSupply == 0
```

Comment:
Let's have regular 

```
if () { 
  ...
} else if () {
  ...
} else {
  ...
}
```

Instead of nested ifs

Reply:
done